### DISCUSSION TITLE
I'm in database restore hell.  Nothing I do seems to cause the database to reload

### DISCUSSION THREAD

**Discussion by TheGorf** [mastodon/mastodon#25030] [Q&A]
I have a complete backup of my Postgres database. I can open it, see it, all the data is there.  When I go in to restore it I'm doing this:

docker cp postgres.dump  db:/postgres.dump 
docker exec -t db createdb -U postgres -T template0
docker exec -t db pg_restore -U postgres -n public --no-owner -d postgres --verbose /postgres.dump

This produces plenty of output but a ton of errors about foreign key constraints. It claims at the end that the errors can be ignored? I can see the data is in the database. But the Mastodon service just acts like it's staring at an empty database. I can not get this thing to restore at all and it's maddening. I'm about to just burn the damn thing to the ground and tell my users they just have to recreate their profiles. What am I doing wrong??

**Comment by TheGorf** [comment]
Anyone?

**Comment by justdaj** [comment]
My observation would be that you're not creating the Mastodon dB or restoring as the Mastodon user

createdb -T template0 mastodon_production
pg_restore -Fc -U mastodon -n public --no-owner --role=mastodon \
  -d mastodon_production backup.dump

I've just completed a successful server migration, which worked, so that the extent of my knowledge on dB importing

**Comment by TheGorf** [comment]
I would be interested to know what your backup commands look like.  Because I am finding more and more sites that suggest the Mastodon documentation is far too limited and should be using the --inserts with the pg_dump/pg_dumpall commands. I have been testing that and it does indeed fix all the foreign key constraint issues.

**Reply by justdaj** [reply]
I used the following to backup and restore

Dump the DB logged in/su as the mastodon user
`
pg_dump -Fc mastodon_production -f backup.dump`

For the new SQL server, as a root user

`sudo -u postgres psql`

```
CREATE USER mastodon CREATEDB;
\q
```

login/su as the mastodon user, create a blank DB and import the dump from earlier

`created -T template0 mastodon_production`

`pg_restore -Fc -U mastodon -n public --no-owner --role=mastodon -d mastodon_production backup.dump`
